fmriprep/utils/transforms.py

```diff
@@ -0,0 +1,92 @@
+"""Utilities for loading transforms for resampling"""
+from pathlib import Path
+
+import h5py
+import nibabel as nb
+import nitransforms as nt
+import numpy as np
+from nitransforms.io.itk import ITKCompositeH5
+
+
+def load_transforms(xfm_paths: list[Path], inverse: list[bool]) -> nt.base.TransformBase:
+    """Load a series of transforms as a nitransforms TransformChain
+
+    An empty list will return an identity transform
+    """
+    if len(inverse) == 1:
+        inverse = inverse * len(xfm_paths)
+    elif len(inverse) != len(xfm_paths):
+        raise ValueError("Mismatched number of transforms and inverses")
+
+    chain = None
+    for path, inv in zip(xfm_paths[::-1], inverse[::-1]):
+        path = Path(path)
+        if path.suffix == '.h5':
+            xfm = load_ants_h5(path)
+        else:
+            xfm = nt.linear.load(path)
+        if inv:
+            xfm = ~xfm
+        if chain is None:
+            chain = xfm
+        else:
+            chain += xfm
+    if chain is None:
+        chain = nt.base.TransformBase()
+    return chain
+
+
+def load_ants_h5(filename: Path) -> nt.TransformChain:
+    """Load ANTs H5 files as a nitransforms TransformChain"""
+    affine, warp, warp_affine = parse_combined_hdf5(filename)
+    warp_transform = nt.DenseFieldTransform(nb.Nifti1Image(warp, warp_affine))
+    return nt.TransformChain([warp_transform, nt.Affine(affine)])
+
+
+def parse_combined_hdf5(h5_fn, to_ras=True):
```

Comment:
i'm confused by the assertions in this - are we just comparing with the hardcoded ants parameters? if so, are there difference if run sloppy?

Reply:
I have not noticed a difference in sloppy. The assertions are to make sure that we are not applying these modifications to files that they don't apply to.

Reply:
@feiling could probably answer better than I could.

Reply:
I wasn't sure if my understanding of the h5 file content format is correct.  It was mainly based on my reverse engineering.

I tested the function extensively for T1 -> MNI transforms, and I am very sure it works properly in this case.  I haven't tested it in other conditions.